src/core/storage/state-migrations.ts

```diff
@@ -68,20 +69,41 @@ export async function migrateWorkspaceToGlobalStorage(context: vscode.ExtensionC
 
 export async function migrateTaskHistoryToFile(context: vscode.ExtensionContext) {
 	try {
-		// If the old taskHistory vs code global state is undefined, do nothing
-		const vscodeGlobalStateTaskHistory = await context.globalState.get("taskHistory")
-		if (vscodeGlobalStateTaskHistory === undefined) {
-			return
+		// Get data from both locations
+		const vscodeGlobalStateTaskHistory = (await context.globalState.get("taskHistory")) as HistoryItem[] | undefined
+		const newLocationData = await readTaskHistoryFromState(context)
+
+		// Normalize old location data to array
+		const oldLocationData = Array.isArray(vscodeGlobalStateTaskHistory) ? vscodeGlobalStateTaskHistory : []
+
+		console.log("[Storage Migration] taskHistory from old location (vscode global state): ", vscodeGlobalStateTaskHistory)
+		console.log("[Storage Migration] taskHistory from new location (file): ", newLocationData)
+
```

Comment:
We could exit early if no migration is needed so we could skip all the `oldLocationData.length > 0` check below to simply the if else conditions

```suggestion
		// Early return if no migration needed
		if (oldLocationData.length === 0) {
			console.log("[Storage Migration] No task history to migrate")
			return
		}
```